networks/p2p/dial.go

```diff
@@ -564,6 +572,7 @@ func (t *dialTask) dial(srv Server, dest *discover.Node) error {
 		return &dialError{err}
 	}
 	mfd := newMeteredConn(fd, false)
+	// TODO close failed fd if needed
```

Comment:
I think we do close `fd` in `c.close(err)` in `setupConn` method.

Reply:
Thank you. You are right :)
I erased it.